projects/gloo/api/v1/options/tracing/tracing.proto

```diff
@@ -97,3 +104,36 @@ message TracingTagLiteral {
     // Static literal value to populate the tag value.
     google.protobuf.StringValue value = 2;
 }
+
+// Requests can produce traces with custom tags.
+// TracingTagLiteral defines a literal which gets added as custom tag.
+message TracingTagMetadata {
+    enum Kind {
+      // The metadata is extracted from the stream metadata.
+      REQUEST = 0;
+      // The metadata is extracted from the endpoint metadata.
+      ENDPOINT = 1;
+    }
+    // Used to populate the tag name.
+    string tag = 1;
+    // The kind of metadata to extract the value from 
+    Kind kind = 2;
+
+    message MetadataValue {
+      // The key to extract the value from the metadata.
+      // If empty will default to JWT principal namespace.
+      string namespace = 1;
+      // The key to extract the value from the metadata.
+      // For example, `principal.iss` or `principal.sub` to extract those claims from the JWT ns
+      string key = 2;
```

Comment:
If the `key` isn't set, is just the namespace used? Or are both `key` and `namespace` required?

Reply:
Both need to be set, there's no default NS here since it really depends on the use-case

Reply:
Makes sense, we might want to highlight this in the docs/proto comments somehow in the follow up